lib/uri.js

```diff
@@ -90,6 +93,10 @@ URI.prototype.toString = function(options) {
                 // Omit optional segment & return
                 return uriStr;
             }
+        } else if (this._pathMetadata
+                && this._pathMetadata[i]
+                && this._pathMetadata[i] === '+') {
+            uriStr += '/' + utils.encodeReserved(segment);
```

Comment:
Yeah, you are right. There are still some chars (like space) that ought to be escaped. You might be able to get that effect more efficiently with `encodeURI`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI

> encodeURI replaces all characters except the following with the appropriate UTF-8 escape sequences:

> Type	Includes
> Reserved characters	; , / ? : @ & = + $
> Unescaped characters	alphabetic, decimal digits, - _ . ! ~ * ' ( )
> Score	#